src/Components/Trailer/Trailer.jsx

```diff
@@ -0,0 +1,35 @@
+import {useParams} from 'react-router-dom'
+import {useDispatch, useSelector} from 'react-redux'
+import React, {useEffect} from 'react'
+import {setTrailerKey} from '../../redux/headerReducer'
+import {getTrailerKey} from '../../redux/selectors/selectors'
+import styles from './Trailer.module.scss'
+
+const Trailer = ({callback, movieId}) => {
+  const dispatch = useDispatch()
+  const id = useParams().id || movieId
+  useEffect(() => {
+    dispatch(setTrailerKey(id))
+  }, [id])
+  const trailerKey = useSelector(getTrailerKey)
+
+  return (
+      <div
+          className={styles.iframeContainer}
+          onClick={() => callback(false)}
+      >
+        {trailerKey
```

Comment:
Let's move it outside the component like 

`trailerKey && <Trailer />`

another way to use some condition here like 

`trailerKey ? <iframe ... > : 'Trailer is not found';`